### ISSUE TITLE
What is the intended behavior of authState.isPending? 

### ISSUE THREAD

**Issue by lee-bennie** [okta/okta-react#14]
I'm having a hard time understanding how `authState.isPending` is supposed to behave.  

When I'm at @okta/okta-react@3.0.3, and look at `authState.isPending` in the Security component, i can see that it's true until isAuthenticated resolves to true or false.  That makes sense to me. 

However, at @okta/okta-react@3.0.4 and above, `authState.isPending` stays true after `isAuthenticated` has resolved to false. This is with a clean slate.  Nothing okta-related in localStorage, session storage, cookies, or anything.  

So I'm wondering which one of these is the intended behavior, as they are very different.  Is this a bug? It sort of contradicts this: https://developer.okta.com/code/react/okta_react/#connect-the-routes, as `isPending` doesn't change.

**Comment by fetters5** [comment]
To confirm you tried with 3.0.7?  I know I had issues with pending in versions 3.04 to 3.06

**Comment by lee-bennie** [comment]
> To confirm you tried with 3.0.7? I know I had issues with pending in versions 3.04 to 3.06

Yeah that's correct.   Tried 3.0.0, 3.0.1, 3.0.2, 3.0.3, all working as expected.  Tried 3.0.4, 3.0.5, 3.0.6, 3.0.7, 3.0.8, all not working as expected.

**Comment by swiftone** [comment]
> it's true until isAuthenticated resolves to true or false. That makes sense to me.

This is indeed how isPending is intended to behave (the info to decide if isAuthenticated is true is async, so isPending indicates if we're waiting for that decision)

> authState.isPending stays true after isAuthenticated has resolved to false

This should not happen - can you share some code that is triggering this for you?

**Comment by lee-bennie** [comment]
> should not happen - can you share some code that is triggering this for you?

After digging deeper and stripping out some stuff in order to share code to reproduce, it turns out that this is related to another dependency that is only conflicting with the mentioned versions of okta-react.  So I think I'm good here.  Sorry for the hassle!

**Comment by lee-bennie** [comment]
Ok sorry to reopen, but here's some code that actually is reproducing the problem on `3.0.4` and above. 

```
import React from 'react';
import ReactDOM from 'react-dom';
import {
  Security,
  LoginCallback,
  SecureRoute,
  useOktaAuth,
} from '@okta/okta-react';

import { BrowserRouter as Router, useHistory, Route } from 'react-router-dom';

const MOUNT_NODE = document.getElementById('app');

const oktaConfig = {
  clientId: `${process.env.OKTA_CLIENT_ID}`,
  issuer: `${process.env.OKTA_ISSUER}`,
  redirectUri: `${window.location.origin}/login/callback`,
  scopes: ['openid', 'profile', 'email'],
  pkce: true,
};

const Home = () => {
  return <div>home</div>;
};

const SomethingProtected = () => {
  return <div>something protected</div>;
};

const LoginPage = () => {
  const { authState } = useOktaAuth();
  return <div>{JSON.stringify(authState)}</div>;
};

const AppWithRouterAccess = () => {
  const history = useHistory();
  const onAuthRequired = () => {
    history.push('/login');
  };

  return (
    <Security {...oktaConfig} onAuthRequired={onAuthRequired}>
      <SecureRoute exact path="/" component={Home} />
      <SecureRoute path="/something-protected" component={SomethingProtected} />
      <Route path="/login" render={() => <LoginPage />} />
      <Route path="/login/callback" component={LoginCallback} />
    </Security>
  );
};

const render = () => {
  ReactDOM.render(
    <Router>
      <AppWithRouterAccess />
    </Router>,
    MOUNT_NODE,
  );
};

render();
```

**Comment by lee-bennie** [comment]
Just want to provide more steps to reproduce: on https://github.com/okta/samples-js-react/tree/master/custom-login. 

Click the login button to navigate to /login.  
`authState.isPending` is now true.

**Comment by swiftone** [comment]
@lee-bennie - We think we've tracked this down and have a fix on the way, but it's been a bit delayed.

If you want to patch your local version of okta-react in the meantime, here's the PR: https://github.com/okta/okta-react/pull/17

**Comment by lee-bennie** [comment]
> @lee-bennie - We think we've tracked this down and have a fix on the way, but it's been a bit delayed.
> 
> If you want to patch your local version of okta-react in the meantime, here's the PR: #17

Ahhh great. Thanks so much!

**Comment by kbobrowski** [comment]
@swiftone thanks for the fix, bumping to 3.0.10 solved this issue for me. You may want to bump the version in your tutorial (https://developer.okta.com/code/react/okta_react/, in `This guide is for @okta/okta-auth-js v4.0.2 and @okta/okta-react v3.0.8.`), as this example is broken in 3.0.8

**Comment by shuowu** [comment]
@kbobrowski Thanks for the suggestion! We do have tasks to update the docs, but it will target the latest okta-react version 4.0 instead of 3.0.10.
